src/main/java/com/conveyal/gtfs/api/graphql/RouteFetcher.java

```diff
@@ -44,6 +45,27 @@
         return routes;
     }
 
+    public static List<WrappedGTFSEntity<Route>> fromStop(DataFetchingEnvironment environment) {
+        WrappedGTFSEntity<Stop> stop = (WrappedGTFSEntity<Stop>) environment.getSource();
+        List<String> routeIds = environment.getArgument("route_id");
+
+        FeedSource feed = ApiMain.getFeedSource(stop.feedUniqueId);
+        List<WrappedGTFSEntity<Route>> routes = feed.feed.patterns.values().stream()
+                .filter(p -> p.orderedStops.contains(stop.entity.stop_id))
+                .map(p -> feed.feed.routes.get(p.route_id))
```

Comment:
You're going to wind up with duplicate routes here. Maybe after the filter do `.map(p -> p.route_id).distinct().map(feed.feed.routes::get)`